common/src/test/java/org/opensearch/ml/common/input/nlp/TextSimilarityMLInputTest.java

```diff
@@ -0,0 +1,134 @@
+        parser.nextToken();
+
+        MLInput parsedInput = MLInput.parse(parser, input.getFunctionName().name());
+        assert (parsedInput instanceof TextSimilarityMLInput);
+        TextSimilarityMLInput parsedTSMLI = (TextSimilarityMLInput) parsedInput;
+        List<String> docs = ((TextSimilarityInputDataSet) parsedTSMLI.getInputDataset()).getTextDocs();
+        String queryText = ((TextSimilarityInputDataSet) parsedTSMLI.getInputDataset()).getQueryText();
+        assert (docs.size() == 2);
+        assert (docs.get(0).equals("That is a happy dog"));
+        assert (docs.get(1).equals("it's summer"));
+        assert (queryText.equals("today is sunny"));
+    }
+
+    @Test
+    public void testXContent_String() throws IOException {
+        XContentBuilder builder = MediaTypeRegistry.contentBuilder(XContentType.JSON);
+        input.toXContent(builder, ToXContent.EMPTY_PARAMS);
+        String jsonStr = builder.toString();
+        assert (jsonStr.equals("{\"algorithm\":\"TEXT_SIMILARITY\",\"query_text\":\"today is sunny\",\"text_docs\":[\"That is a happy dog\",\"it's summer\"]}"));
+    }
+
+    @Test
+    public void testParseJson() throws IOException {
+        String json = "{\"algorithm\":\"TEXT_SIMILARITY\",\"query_text\":\"today is sunny\",\"text_docs\":[\"That is a happy dog\",\"it's summer\"]}";
+        XContentParser parser = XContentType.JSON.xContent()
+                .createParser(new NamedXContentRegistry(new SearchModule(Settings.EMPTY,
+                        Collections.emptyList()).getNamedXContents()), null, json);
+        parser.nextToken();
+
+        MLInput parsedInput = MLInput.parse(parser, input.getFunctionName().name());
+        assert (parsedInput instanceof TextSimilarityMLInput);
+        TextSimilarityMLInput parsedTSMLI = (TextSimilarityMLInput) parsedInput;
+        List<String> docs = ((TextSimilarityInputDataSet) parsedTSMLI.getInputDataset()).getTextDocs();
+        String queryText = ((TextSimilarityInputDataSet) parsedTSMLI.getInputDataset()).getQueryText();
+        assert (docs.size() == 2);
+        assert (docs.get(0).equals("That is a happy dog"));
+        assert (docs.get(1).equals("it's summer"));
+        assert (queryText.equals("today is sunny"));
+    }
+
+    @Test
+    public void testParseJson_NoPairs_ThenFail() throws IOException {
+        String json = "{\"algorithm\":\"TEXT_SIMILARITY\",\"query_text\":\"today is sunny\",\"text_docs\":[]}";
+        XContentParser parser = XContentType.JSON.xContent()
+                .createParser(new NamedXContentRegistry(new SearchModule(Settings.EMPTY,
+                        Collections.emptyList()).getNamedXContents()), null, json);
+        parser.nextToken();
+
+        IllegalArgumentException e = assertThrows(IllegalArgumentException.class,
+            () -> MLInput.parse(parser, input.getFunctionName().name()));
+        assert (e.getMessage().equals("no text docs"));
```

Comment:
Can we have another test for missing query text too?

Reply:
sure